java/clients/src/test/java/sleeper/status/report/compactionjob/CompactionJobStatusCollectorTest.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright 2022 Crown Copyright
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package sleeper.status.report.compactionjob;
+
+import org.junit.Test;
+import sleeper.compaction.job.CompactionJobStatusStore;
+import sleeper.compaction.job.status.CompactionJobStatus;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+import static sleeper.status.report.compactionjob.StatusReporterTestBase.jobWithMultipleRuns;
+
+public class CompactionJobStatusCollectorTest {
+    private static final String JOB_ID_FOUND = "job-id-exists";
+    private static final String JOB_ID_NOT_FOUND = "job-id-not-found";
+    private final CompactionJobStatusStore statusStore = mock(CompactionJobStatusStore.class);
+    private final CompactionJobStatusCollector collector = new CompactionJobStatusCollector(statusStore, "table-test");
+
+    private CompactionJobStatus getJob() {
+        return jobWithMultipleRuns().get(0);
```

Comment:
Maybe use a job that's been created but not started? It seems simpler to avoid unnecessary data. It might be nice to show that the ID is what we expect.